interfaces/hotplug/spec.go

```diff
@@ -34,31 +34,30 @@ type Definer interface {
 
 // SlotSpec is a definition of the slot to create in response to udev event.
 type SlotSpec struct {
-	// XXX: Name is the name the interface wants to give to the slot; we
-	// might want to mediate this though (e.g. generate automatically), so this
-	// may change/go away.
+	// Name is the name the interface wants to give to the slot. It can be
+	// left empty, in such case hotplug machinery will auto-generate a name.
+	// If the spec provides a name, it needs to be a syntactically valid
+	// slot name, hotplug machinery will only make sure it's a unique name
+	// by appending numeric suffix if needed.
 	Name  string
 	Label string
 	Attrs map[string]interface{}
 }
 
-// Specification contains data about all slots that a hotplug interface wants to have created in response to uevent.
+// Specification contains a slot definition to create in response to uevent.
 type Specification struct {
-	// slots are indexed by slot name to ensure unique names
-	slots map[string]*SlotSpec
+	slot *SlotSpec
 }
 
 // NewSpecification creates an empty hotplug Specification.
 func NewSpecification() *Specification {
-	return &Specification{
-		slots: make(map[string]*SlotSpec),
-	}
+	return &Specification{}
 }
 
-// AddSlot adds a specification of a slot.
-func (h *Specification) AddSlot(slotSpec *SlotSpec) error {
-	if _, ok := h.slots[slotSpec.Name]; ok {
-		return fmt.Errorf("slot %q already exists", slotSpec.Name)
+// SetSlot adds a specification of a slot.
+func (h *Specification) SetSlot(slotSpec *SlotSpec) error {
+	if h.slot != nil {
+		return fmt.Errorf("slot specification already created")
 	}
 	if err := snap.ValidateSlotName(slotSpec.Name); err != nil {
```

Comment:
The comment around SlotSpec suggest that Name can be empty and hotplug will assign something to it in such case. Will this error out if slotSpec.Name is empty?

Reply:
Good catch, thanks! Fixed.